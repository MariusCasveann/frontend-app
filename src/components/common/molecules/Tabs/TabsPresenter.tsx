import React from 'react';
import { Tabs } from 'antd';

const { TabPane } = Tabs;

export interface TabProps {
    tabContent: React.Component | string | any;
    tabName: string;
}

export interface TabsProps {
    tabsData: TabProps[];
    defaultActiveKey?: string;
}

export default ({ tabsData, defaultActiveKey = '0' }: TabsProps) => (
    <Tabs defaultActiveKey={defaultActiveKey} type="card">
        {tabsData.map((item, index) => (
            <TabPane tab={item.tabName} key={index.toString()}>
                {typeof item.tabContent === 'string' ? item.tabContent : <item.tabContent />}
            </TabPane>
        ))}
    </Tabs>
);
