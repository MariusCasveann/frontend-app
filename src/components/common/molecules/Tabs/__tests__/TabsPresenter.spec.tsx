import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';
jest.doMock('antd', () => ({ Tabs: 'Tabs', TabPane: 'TabPane' }));
import TabsPresenter from '../TabsPresenter';

describe('TabsPresenter', () => {
    it('renders correctly', () => {
        const tabsData = [
            {
                tabContent: 'content',
                tabName: 'name'
            }
        ];
        const renderer = ShallowRenderer.createRenderer();
        renderer.render(<TabsPresenter tabsData={tabsData} />);
        const result = renderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });
});
