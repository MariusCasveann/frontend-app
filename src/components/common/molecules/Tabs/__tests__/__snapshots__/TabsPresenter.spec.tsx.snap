// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsPresenter renders correctly 1`] = `
<Tabs
  defaultActiveKey="0"
  hideAdd={false}
  tabPosition="top"
  type="card"
>
  <TabPane
    placeholder={null}
    tab="name"
  >
    content
  </TabPane>
</Tabs>
`;
